controllers/components/feastoperator/feastoperator_support.go

```diff
@@ -0,0 +1,38 @@
+package feastoperator
+
+import (
+	conditionsv1 "github.com/openshift/custom-resource-status/conditions/v1"
+
+	componentApi "github.com/opendatahub-io/opendatahub-operator/v2/apis/components/v1alpha1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/controllers/status"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/types"
+	odhdeploy "github.com/opendatahub-io/opendatahub-operator/v2/pkg/deploy"
+)
+
+const (
+	ComponentName = componentApi.FeastOperatorComponentName
+
+	ReadyConditionType = conditionsv1.ConditionType(componentApi.FeastOperatorKind + status.ReadySuffix)
+
+	// LegacyComponentName is the name of the component that is assigned to deployments
+	// via Kustomize. Since a deployment selector is immutable, we can't upgrade existing
+	// deployment to the new component name, so keep it around till we figure out a solution.
+	LegacyComponentName = "feastoperator"
```

Comment:
It's a new component and does not have legacy. I think, you can use just name.

Reply:
Fixed : Removed the `LegacyComponentName` and use `ComponentName`